### PR TITLE
Improving test coverage

### PR DESCRIPTION
**Pull request by vsukhomlinov**
Fixes https://github.com/diasdavid/js-libp2p/issues/5

### CODE REVIEW

**Comment by diasdavid** [comment]
Seems I added travis after and didn't notice I added 0.4 instead of 4 for the Node.js version. Mind rebasing so that travis runs the tests again?

https://github.com/diasdavid/js-multistream/commit/7125d9345d9310bcd6b8fe3f012bc451819b9a13

File: tests/impl/one-way-test.js [hunk 1]
```diff
@@ -68,4 +98,35 @@ experiment('Node.js Implementation: ', function () {
...
+        done()
+      })
+    }).listen(8021)
+
+    var socket = tcp.connect({port: 8021}, function tcpConnectionOpen () {
+      lpm.write(socket, '/garbage/1.0.0\n')
+    })
+  })
+
+  test('closing socket for unsupported handler', function (done) {
```
Comment (diasdavid) [inline tests/impl/one-way-test.js]: where is the check to see if the socket is closed?

File: tests/impl/one-way-test.js [hunk 2]
```diff
@@ -68,4 +98,35 @@ experiment('Node.js Implementation: ', function () {
...
+    var acc = new MultiStream.Silent()
+    tcp.createServer(function (socket) {
+      acc.handle(socket, function () {
+        acc.addHandler('/none/1.2.3', function (err) {
+          expect(err.message).to.equal('Received non supported Protocol or Version: /none/1.0.0')
```
Comment (diasdavid) [inline tests/impl/one-way-test.js]: It should not be up to the protocol handler each time there is an attempt to handshake a protocol that isn't itself. Several protocol handlers might be added to the same connection

**Comment by diasdavid** [comment]
Hey @vsukhomlinov, thank you for the PR. I'm missing to see how this fixes #5, could you point me to that line?

**Comment by vsukhomlinov** [comment]
It doesn't. It fixes #5 from another repository.

-Vitaly

File: .travis.yml [hunk 3]
```diff
@@ -2,6 +2,15 @@ language: node_js
 node_js:
   - "4"
   - "5"
+env:
+  - CXX=g++-4.8 #need that for the buffertools dependency
```
Comment (diasdavid) [inline .travis.yml]: where is this used?

**Comment by diasdavid** [comment]
Closing in favour of https://github.com/diasdavid/js-multistream/pull/9

Thank you for bringing this @vsukhomlinov

**Pull request closed without merging**